clients/fides-js/src/lib/consent-types.ts

```diff
@@ -74,6 +75,26 @@ export type FidesOptions = {
   // An explicitly passed-in TC string that supersedes the cookie, and prevents any API calls to fetch
   // experiences / preferences. Only available when TCF is enabled. Optional.
   fidesString: string | null;
+
+  // Allows for explicit overrides on various internal API calls made from Fides.
+  api: FidesApiOptions | null;
+};
+
+export type FidesApiOptions = {
+  /**
+   * Intake a custom function that is called instead of the internal Fides API to save user preferences.
+   *
+   * @param {object} Fides  - the global Fides object with all configured options, state, etc.
+   * @param {object} consent - updated version of Fides.consent with the user's saved preferences for Fides notices
+   * @param {string} fides_string - updated version of Fides.fides_string with the user's saved preferences for TC/AC/etc notices
+   * @param {object} experience - current version of the privacy experience that was shown to the user
+   */
+  savePreferencesFn: (
+    fides: FidesConfig,
```

Comment:
This feels odd, why a `FidesConfig` type here? It doesn't match the comment and also is pretty narrow. Can't this be the `Fides` global?

Reply:
`FidesConfig` is the entire obj we use to init Fides in `Fides.init()`. I'm a little hesitant to pass in the `Fides` global as it's something we build during the init and state management could lead to inconsistencies in what we have in `Fides` vs in the cookie, depending on when we check for those vals. But I'll go ahead with refactoring to pass the `Fides` global in for now.

Reply:
Running into some tricky bits here where the `Fides` global isn't updated until _after_ we initialize the overlay:

```
const updatedFides = await initialize({
    cookie,
    _Fides,
    experience,
    renderOverlay,
    updateCookie,
  });
  Object.assign(_Fides, updatedFides);
```

So if we pass it in, it won't be completely up to date, unless we do some refactoring. Additionally there are some cyclical import issues with the `Fides` type as it's defined in `initialize.ts` and it's not as simple as moving it to another types file.